pyanaconda/ui/lib/subscription.py

```diff
@@ -398,7 +391,14 @@ def unregister(payload, overridden_source_type, progress_callback=None, error_ca
                 log.debug("subscription thread: restarting payload after unregistration")
                 _do_payload_restart(payload)
 
-        log.debug("Subscription GUI: unregistration succeeded")
+        # check if we were registered to a Satellite instance and roll that back as well
+        # if needed
+        if subscription_proxy.IsRegisteredToSatellite:
+            log.debug("subscription thread: rolling back Satellite provisioning")
+            roll_back_satellite_provisioning()
```

Comment:
Couldn't we do this in the `UnregisterWithTask` class?

Reply:
Actually, this looks like a leftover as `UnregisterTask` now handles also Satellite provisioning rollback. So removed. :)

Reply:
The code hasn't changed.

Reply:
Should be fixed by the latest push as well.